pontoon/machinery/views.py

```diff
@@ -113,6 +117,30 @@ def machine_translation(request):
         return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
 
 
+def machine_translation_caighdean(request):
+    """Get translation from Caighdean machine translation service."""
+    try:
+        entityid = int(request.GET['id'])
+    except (MultiValueDictKeyError, ValueError) as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+
+    try:
+        entity = Entity.objects.get(id=entityid)
+    except Entity.DoesNotExist as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
```

Comment:
Why not use ``get_object_or_404`` here?

Reply:
probs that would be better, ill update

Reply:
hmm, im also wondering whether the other bad responses, should be JSON

Reply:
So in this particular case (if you go with `get_object_or_404`) `HttpResponseBadRequest` won't be used, right?

In general, I don't know. Is there a good reason to move the output to JSON?

Reply:
this is what i was thinking to do...

```
    try:
        entity = get_object_or_404(Entity, id=entityid)
    except http.Http404 as e:
        return http.HttpResponseNotFound(
            json.dumps(dict(error=404, message=str(e))),
            content_type='application/json')
```

I think the reason it *should* respond with JSON, is because (i figure) the client makes the request with `application/json` as the `Content-Type` and expects that - certainly that is what is returned in the case of success

Reply:
...from a practical point of view, its also a pita when XHR requests get full page errors back

if you are debugging it, you have to sift through the unrendered html to find what the error was. 

In this case, ie simple get_object_or_404 it probs wouldnt matter o much

Reply:
LGTM